app/utils/AuthProvider.tsx

```diff
@@ -0,0 +1,174 @@
+'use client';
+
+import {
+  createContext,
+  ReactNode,
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+import { useRouter } from 'next/navigation';
+import supabase from '../../api/supabase/createClient';
+
+interface AuthContextType {
+  authUser: AuthUser | null;
+  isLoggedIn: boolean;
+  signUp: (email: string, password: string) => Promise<void>;
+  signIn: (email: string, password: string) => Promise<void>;
+  signOut: () => Promise<void>;
+  loading: boolean;
+}
+
+interface AuthUser {
+  id: string;
+  email: string;
+}
+
+// Create the AuthContext
+const AuthContext = createContext<AuthContextType | null>(null);
+
+// Custom hook to use AuthContext
+export function useAuth() {
+  const context = useContext(AuthContext);
+  if (!context) {
+    throw new Error('useAuth must be used within an AuthProvider');
+  }
+  return context;
+}
+
+// AuthProvider component that wraps around your app or specific pages
+export function AuthProvider({ children }: { children: ReactNode }) {
+  const [authUser, setAuthUser] = useState<AuthUser | null>(null);
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [loading, setLoading] = useState(true);
+  const { push } = useRouter(); // Get Next.js router for navigation
+
+  // Sign Up function
+  const signUp = async (email: string, password: string) => {
+    try {
+      const { data, error } = await supabase.auth.signUp({ email, password });
+      if (error) throw new Error(error.message);
+
+      if (data.user) {
+        // Map the user data to AuthUser type, ensuring `id` and `email` are defined
+        const user: AuthUser = {
+          id: data.user.id ?? '', // Fallback to an empty string if `id` is undefined
+          email: data.user.email ?? '', // Fallback to an empty string if `email` is undefined
+        };
+
+        setAuthUser(user);
+        setIsLoggedIn(true);
+        push('/dashboard'); // Redirect to dashboard after sign-up
```

Comment:
the redirecting should be moved to /signup. The functions in AuthProvider should only handle the supabase calls and updating the AuthContext. All additional logic/processing should happen within the respective page

also, since /dashboard is a test page that will be deleted, we should nav to '/' instead (i.e. the home page)